client/my-sites/checkout/upsell-nudge/index.jsx

```diff
@@ -180,10 +192,65 @@ export class UpsellNudge extends React.Component {
 			`calypso_${ upsellType.replace( /-/g, '_' ) }_${ buttonAction }_button_click`
 		);
 
+		if ( this.isEligibleForOneClickUpsellABTest( buttonAction ) ) {
+			this.setState( {
+				showPurchaseModal: true,
+				cartLastServerResponseDate: this.getCartUpdatedTime(),
+			} );
+			replaceCartWithItems( [ this.props.product ] );
+			return;
+		}
+
 		return siteSlug
 			? page( `/checkout/${ upgradeItem }/${ siteSlug }` )
 			: page( `/checkout/${ upgradeItem }` );
 	};
+
+	isEligibleForOneClickUpsellABTest = ( buttonAction ) => {
+		const { cards, upsellType } = this.props;
+
+		if ( 'accept' !== buttonAction || 'concierge-quickstart-session' !== upsellType ) {
+			return false;
+		}
+
+		// stored cards should exist
+		if ( cards.length === 0 ) {
+			return false;
+		}
+
+		return 'test' === abtest( 'oneClickUpsell' );
```

Comment:
If the user has two sites, and one of the sites has a stored card, then there's a scenario in which the purchase model could get stuck. To reproduce:

1. On an account that has two sites, purchase a paid plan on one of the sites using a credit card so that your card gets saved.
2. Visit `/checkout/offer-quickstart-session`. Users would reach this link either via the Quick Start upsell banner in `/me/purchases`, or via HEs upselling a Quick Start session via chat or tickets.
3. Click on the "Yes" button on the upsell and notice the purchase model stuck while loading(assuming you get assigned to the variant, else manually assign yourself to the variant and try again).


![signup](https://user-images.githubusercontent.com/1269602/93999393-61341d00-fdb3-11ea-9b57-29f3812b5414.gif)

Reply:
Thanks @niranjan-uma-shankar ! I think the bug occurs due to lack of the site slug which the modal component depends on. Now the test is disabled when no site is selected. Can you check it again?